src/core/auth/authentication/components/Kratos/KratosForm.tsx

```diff
@@ -0,0 +1,46 @@
+import { UiContainer } from '@ory/kratos-client';
+import { Box, BoxProps } from '@mui/material';
+import { createContext, Dispatch, FormEvent, useCallback, useContext, useMemo, useState } from 'react';
+
+interface KratosFormProps extends BoxProps<'form'> {
+  ui?: UiContainer;
+}
+
+interface KratosFormContextValue {
+  isFormValid: boolean;
+  setIsFormValid: Dispatch<boolean>;
+}
+
+const KratosFormContext = createContext<KratosFormContextValue | null>(null);
+
+const KratosForm = ({ ui, children, ...formProps }: KratosFormProps) => {
+  const [isFormValid, setIsFormValid] = useState(true);
+
+  const handleSubmit = useCallback((e: FormEvent<HTMLFormElement>) => {
+    const button = document.activeElement as HTMLButtonElement | null;
+    // do check if only submitting password method
+    if (button && button.name === 'method' && button.value === 'password') {
```

Comment:
this is reliant on Kratos payload having specific values, it will be nice to extract these values as constants, ideally in a separate file, so we can quickly find them if Kratos response types change in the future